addons/udes_stock/models/stock_picking_type.py

```diff
@@ -660,10 +659,20 @@ def get_action_package_tree(self):
         if not isinstance(action.get("context", 0), dict):
             action["context"] = {}
 
-        action["context"].update({
-            "active_ids": location_ids,
-            "hide_create_date": False,
-        })
+        action["context"].update({"active_ids": location_ids, "hide_create_date": False})
         action["domain"] = [("location_id", "child_of", location_ids)]
 
         return action
+
+    def _get_goods_in_picking_type_for_warehouse(self, company_id, warehouse_id):
+
+        Warehouse = self.env["stock.warehouse"]
+        warehouse = Warehouse.search(
+            [("company_id", "=", company_id), ("id", "=", warehouse_id)]
+        )
+        if not warehouse:
+            raise ValidationError(
+                ("Cannot find warehouse for warehouse id %s , company id %s")
+                % (warehouse_id, company_id)
+            )
+        return warehouse.u_goods_in_picking_type_id
```

Comment:
Just been playing around with this and there's actually already a field that does this for us `in_type_id`, so we don't need `u_goods_in_picking_type_id`